csharp-basic-topics/RemoveIllegalCharsFromPathAndFilenames/RemoveIllegalCharsFromPathAndFilenames/StringMethods.cs

```diff
@@ -0,0 +1,62 @@
+﻿public static class StringMethods
+{
+    public static List<string> CheckForInvalid(List<string> strings, HashSet<char> criteria)
+    {
+        List<string> result = new List<string>();
+
+        foreach (var str in strings)
+        {
+            foreach (var ch in criteria)
+            {
+                if (str.Contains(ch))
+                {
+                    result.Add(str);
+                    break;
+                }
+            }
+        }
+
+        return result;
+    }
+
+    public static List<string> CheckForInvalidLINQ(List<string> strings, HashSet<char> criteria)
+    {
+        return strings.Where(str => str.Any(criteria.Contains)).ToList();
+    }
+
+    public static List<string> CheckForInvalidLINQHeader(List<string> strings, HashSet<char> criteria)
```

Comment:
You can change `List<string>` return type to `IEnumerable<string>` and remove the `ToList()`